core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -175,18 +175,20 @@ object WorkspaceMigrationActor {
       liftF(OptionT.none)
   }
 
-  implicit class MigrateActionOps[A](action: MigrateAction[A]) {
+  implicit class MigrateActionOps[A](fa: => MigrateAction[A]) {
     final def handleErrorWith(f: Throwable => MigrateAction[A]): MigrateAction[A] =
       MigrateAction { env =>
-        OptionT(action.run(env).value.handleErrorWith(f(_).run(env).value))
+        OptionT(fa.run(env).value.handleErrorWith(f(_).run(env).value))
       }
+    final def |(fb: => MigrateAction[A]): MigrateAction[A] =
+      MigrateAction(env => fa(env).orElse(fb(env)))
   }
 
   // Read workspace migrations in various states, attempt to advance their state forward by one
   // step and write the outcome of each step to the database.
   final def migrate: MigrateAction[Unit] =
     List(
-      startMigration,
+      restartMigration | startMigration,
```

Comment:
Preferring restarting migrations over starting new ones ensures we don't saturate the pipeline and leave retries to the end